client/web/src/enterprise/app/settings/AppSettingsArea.tsx

```diff
@@ -0,0 +1,134 @@
+
+import { mdiGit, mdiFolderMultiplePlusOutline } from '@mdi/js'
+import { Routes, Route, Outlet, Navigate, useLocation } from 'react-router-dom'
+
+import { TelemetryProps } from '@sourcegraph/shared/src/telemetry/telemetryService'
+import { Button, H2, Link, Icon, PageHeader } from '@sourcegraph/wildcard'
+
+import { LocalRepositoriesStep, RemoteRepositoriesStep } from '../../../setup-wizard/components'
+
+import styles from './AppSettingsArea.module.scss'
+
+enum AppSettingURL {
+    LocalRepositories = 'local-repositories',
+    ConnectToSourcegraphDotCom = 'connect-to-dot-com',
+    RemoteRepositories = 'remote-repositories',
+}
+
+interface AppSettingsAreaProps extends TelemetryProps {}
+
+export const AppSettingsArea: FC<AppSettingsAreaProps> = props => {
+    const { telemetryService } = props
+
+    return (
+        <Routes>
+            <Route path="*" element={<AppSettingsLayout />}>
+                <Route
+                    path={AppSettingURL.LocalRepositories}
+                    element={<LocalRepositoriesTab telemetryService={telemetryService} />}
+                />
+                <Route
+                    path={`${AppSettingURL.RemoteRepositories}/*`}
+                    element={<RemoteRepositoriesTab telemetryService={telemetryService} />}
+                />
+                <Route
+                    path={AppSettingURL.ConnectToSourcegraphDotCom}
+                    element={<H2>Hello Connect to sourcegraph.com</H2>}
+                />
+
+                <Route path="*" element={<Navigate to={AppSettingURL.LocalRepositories} replace={true} />} />
+            </Route>
+        </Routes>
+    )
+}
+
+interface AppSetting {
+    url: AppSettingURL
+    name: string
+}
+
+const APP_SETTINGS: AppSetting[] = [
+    { url: AppSettingURL.LocalRepositories, name: 'Local repositories' },
+    { url: AppSettingURL.RemoteRepositories, name: 'Remote repositories' },
+    { url: AppSettingURL.ConnectToSourcegraphDotCom, name: 'Connect sourcegraph.com' },
+]
+
+const AppSettingsLayout: FC = () => {
+    const location = useLocation()
+
+    return (
+        <div className={styles.root}>
+            <ul className={styles.navigation}>
+                {APP_SETTINGS.map(setting => (
+                    <li key={setting.url}>
+                        <Button
+                            as={Link}
+                            to={`../${setting.url}`}
+                            variant={location.pathname.includes(`/${setting.url}`) ? 'primary' : undefined}
+                            className={styles.navigationItemLink}
+                        >
+                            {setting.name}
+                        </Button>
+                    </li>
+                ))}
+            </ul>
+
+            <Outlet />
+        </div>
+    )
+}
+
+interface LocalRepositoriesTabProps extends TelemetryProps {}
+
+const LocalRepositoriesTab: FC<LocalRepositoriesTabProps> = props => {
+    const { telemetryService } = props
```

Comment:
suggestion: Simplify code, if we don't foresee adding more props in the very-near-future, I don't think we should have empty interface with only one parent.

```suggestion
const LocalRepositoriesTab: FC<TelemetryProps> = ({ telemetryService }) => {
```